### PR TITLE
Stay on current page when claiming/unclaiming

### LINKED ISSUE #2309: Submission Queue - clicking "claim" changes the sort
From @smcgregor:
> Clicking claim changes the sort, which tends to make the thing you just claimed disappear

### LINKED ISSUE #2310: Submission Queue - Easy "unclaim" button
Nice to have: after an editor "claims" a submission, adding a live "unclaim" button immediately in its place.

(h/t @smcgregor)

### PR DESCRIPTION
**Pull request by clari182**
- Stays on current page when claiming or unclaiming
- Adds unclaim functionality

- closes #2309 
- closes #2310

### CODE REVIEW

File: site/gatsby-site/react-table-config.d.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
+import {
```
Comment (pdcp1) [inline site/gatsby-site/react-table-config.d.ts:1]: Is this file needed? I'm curious about the file path `site/gatsby-site`. It looks inappropriate

**Review by pdcp1 — CHANGES_REQUESTED** [review]

File: site/gatsby-site/react-table-config.d.ts (hunk 1 above)
Comment (clari182) [reply site/gatsby-site/react-table-config.d.ts:1]: Yes, it's to avoid [this](https://stackoverflow.com/questions/60006656/react-table-not-assignable-to-type-partialtablestateobject) issue, following the solution provided by the package [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-table/Readme.md).

Comment (pdcp1) [reply site/gatsby-site/react-table-config.d.ts:1]: OK, I got it! I wonder if we can watch that issue in the `react-table` for a possible official fix.

**Review by pdcp1 — APPROVED** [review]

**Comment by clari182** [comment]
@cesarvarela can you review this PR please? Thanks!

File: site/gatsby-site/react-table-config.d.ts (hunk 1 above)
Comment (cesarvarela) [reply site/gatsby-site/react-table-config.d.ts:1]: I agree with the weird location comment, I think the convention is to have them inside a `/types` folder.

**Review by cesarvarela — COMMENTED** [review]
I think this change is big enough to update or add a new test.

**Comment by pdcp1** [comment]
@clari182 I think we should update this branch with the Staging code to see if the tests pass.
